src/Compilers/Core/Portable/Symbols/IMethodSymbolInternal.cs

```diff
@@ -24,5 +26,7 @@ internal interface IMethodSymbolInternal : ISymbolInternal
         /// <param name="typeArguments">The immediate type arguments to be replaced for type
         /// parameters in the method.</param>
         IMethodSymbolInternal Construct(params ITypeSymbolInternal[] typeArguments);
+
+        ITypeSymbolInternal GetParameterType(int index);
```

Comment:
>ITypeSymbolInternal GetParameterType(int index); [](http://example.com/codeflow?start=8&length=48)

I think that instead we should return immutable array of ```IParameterSymbolInternal``` adding necessary APIs to that interface.